Unexpected `await` inside a loop

C:\Users\lukas\workspace\snake_js\index.js
  12:3   error  Unexpected `await` inside a loop                                                                                                                                         
  no-await-in-loop

--- a/index.js
+++ b/index.js
@@ -1,7 +1,8 @@
 import Snake from "./src/Snake.js";
 
 const snake = new Snake(10, 10);
-for (;;) {
+
+async function main() {
   snake.boardGame.clear();
   const snakeToDraw = snake.getFullSnake();
   for (let i = 0; i < snakeToDraw.length; i += 1) {
@@ -11,7 +12,9 @@ for (;;) {
   console.clear();
   snake.boardGame.printGame(snake.points);
   await snake.changeDirection();
-  if (snake.keyboard.keyValue === "\u001b") {
-    break;
+  if (snake.keyboard.keyValue !== "\u001b") {
+    main();
   }
 }
+
+await main();

--- a/src/BoardGameConsole.js
+++ b/src/BoardGameConsole.js
@@ -30,6 +30,7 @@ export default class BoardGameConsole {
         .fill("-")
         .join("")
     );
+    console.log(`Exit: ESC`);
   }
 
   drawPixel(pixel) {

--- a/src/Keyboard.js
+++ b/src/Keyboard.js
@@ -18,8 +18,8 @@ export async function readStdinSync() {
     process.stdin.setEncoding("utf8");
     process.stdin.resume();
     process.stdin.once("data", (key) => {
-      process.stdin.pause();
       resolve(key);
+      process.stdin.pause();
     });
   });
 }

